rocketpy/Dispersion.py

```diff
@@ -87,841 +101,209 @@ def __init__(
+        # TODO: add pickle package to deal with this and also Function objects
+        inputs_log = []
+        # Loop through each line in the file
+        for line in self.dispersion_input_file:
+            # swap "<" and ">" to \"
+            # this is a way to interpret the trigger functions
+            line = line.replace("<", "'")
+            line = line.replace(">", "'")
+            # # Skip comments lines
+            if line[0] != "{":
+                continue
+            # Try to convert the line to a dictionary
+            d = ast.literal_eval(line)
+            # If successful, append the dictionary to the list
+            inputs_log.append(d)
+
+        return inputs_log
+
+    @cached_property
+    def outputs_log(self):
+        """Save outputs_log log from a file into an attribute for easy access"""
+        outputs_log = []
+        # Loop through each line in the file
+        for line in self.dispersion_output_file:
+            # Skip comments lines
+            if line[0] != "{":
+                continue
+            # Try to convert the line to a dictionary
+            d = ast.literal_eval(line)
+            # If successful, append the dictionary to the list
+            outputs_log.append(d)
+
+        return outputs_log
+
+    @cached_property
+    def errors_log(self):
+        """Save errors_log log from a file into an attribute for easy access"""
+        errors_log = []
+        # Loop through each line in the file
+        for line in self.dispersion_error_file:
+            # Skip comments lines
+            if line[0] != "{":
+                continue
+            # Try to convert the line to a dictionary
+            d = ast.literal_eval(line)
+            # If successful, append the dictionary to the list
+            errors_log.append(d)
 
-    def __process_class_from_dict(
-        self,
-        dictionary,
-        object,
-        class_name,
-        class_name_pretty1,
-        class_name_pretty2=None,
-    ):
-        """Check if all the relevant inputs for the classes are present in
-        the dispersion dictionary, input the missing ones and return the modified
-        dictionary.
+        return errors_log
 
-        Parameters
-        ----------
-        dictionary :  dict
-            Dictionary containing the parameters to be varied in the dispersion
-            simulation. The keys of the dictionary are the names of the parameters
-            to be varied, and the values can be either tuple or list. If the value
-            is a single value, the corresponding class of the parameter need to
-            be passed on the run_dispersion method.
-        object : Environment, SolidMotor, Rocket, Flight
-            Object that will be used to get attributes from.
-        class_name : string
-            Name of the class used in input_dict.
-        class_name_pretty : string
-            Name of the class used for prints.
+    @cached_property
```

Comment:
If I run simulations in append='True' mode, the number of loaded_sims should be modified, no? In that case, does it still seem to be a good idea keeping this variable as a cached_property?

Reply:
Fixed in https://github.com/RocketPy-Team/RocketPy/pull/336/commits/4e8319e6bbaa1ef7afef23386fe4b5076ba9c78f